rfq/oracle.go

```diff
@@ -52,26 +51,15 @@ func (o *OracleError) Error() string {
 	return fmt.Sprintf("OracleError(code=%d, msg=%s)", o.Code, errMsg)
 }
 
-// OracleAskResponse is a struct that holds the price oracle's suggested ask
-// price for an asset.
-type OracleAskResponse struct {
-	// AskPrice is the asking price of the quote.
-	AskPrice *lnwire.MilliSatoshi
+// OracleResponse is a struct that holds the price oracle's suggested buy or
+// sell price for an asset swap.
+type OracleResponse struct {
+	// AssetRate is the asset to BTC rate. Other asset in the transfer is
+	// assumed to be BTC and therefore not included in the response.
+	AssetRate rfqmsg.BigIntFixedPoint
 
-	// Expiry is the price expiryDelay lifetime unix timestamp.
-	Expiry uint64
-
-	// Err is an optional error returned by the price oracle service.
-	Err *OracleError
-}
-
-// OracleBidResponse is a struct that holds the price oracle's suggested bid
-// price for an asset.
-type OracleBidResponse struct {
```

Comment:
So we've unified the bid vs ask structs to simplify? Would a type alias still be useful in certain contexts? So one for bid and the other for asks.

Reply:
> So we've unified the bid vs ask structs to simplify?

Yes. The caller knows what they'll get back because the function that they call is specific to bid/ask.

> Would a type alias still be useful in certain contexts? So one for bid and the other for asks.

I don't think there's much use for an alias right now. Usually we pass the returned value straight into a request or accept with a field name that indicates bid/ask.